server/games/ladder_game.py

```diff
@@ -19,7 +19,22 @@ def __init__(self, id, *args, **kwargs):
     async def rate_game(self):
         if self.validity == ValidityState.VALID:
             new_ratings = self.compute_rating(rating='ladder')
+
             await self.persist_rating_change_stats(new_ratings, rating='ladder')
+            await self._update_division_scores()
+
+    async def _update_division_scores(self):
+        for player in self.players:
+            army = self.get_player_option(player.id, 'Army')
+            try:
+                if self.get_army_score(army) == 1:
+                    winner = player
+                if self.get_army_score(army) == 0:
+                    loser = player
+            except KeyError:
+                return
+
+            self.game_service.game_stats_service.process_ladder_scores(winner, loser, 1)
```

Comment:
No it was not intended. Of course that needs to be outside the for-loop.
But you are right about the draw. I should only proccess scores `if winner is not None`